library/src/main/java/com/woxthebox/draglistview/ColumnProperties.java

```diff
@@ -0,0 +1,218 @@
+
+import androidx.annotation.ColorInt;
+import androidx.annotation.NonNull;
+import androidx.annotation.Nullable;
+import androidx.recyclerview.widget.LinearLayoutManager;
+import androidx.recyclerview.widget.RecyclerView;
+import androidx.recyclerview.widget.RecyclerView.LayoutManager;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Properties of adding/inserting column to the {@link BoardView}
+ *
+ * Instance is created using an associated {@link Builder} object by
+ * invoking its {@link Builder#build build} method.
+ */
+public class ColumnProperties {
+
+    private DragItemAdapter mDragItemAdapter;
+    private LayoutManager mLayoutManager;
+    private List<RecyclerView.ItemDecoration> mItemDecorations;
+    private boolean mHasFixedItemSize;
+    private int mColumnBackgroundColor;
+    private int mItemsSectionBackgroundColor;
+    private View mHeader;
+    private View mColumnDragView;
+
+    private ColumnProperties(DragItemAdapter adapter,
+                             LayoutManager layoutManager,
+                             List<RecyclerView.ItemDecoration> itemDecorations,
+                             boolean hasFixedItemSize,
+                             int columnBackgroundColor,
+                             int itemsSectionBackgroundColor,
+                             View columnDragView,
+                             View header) {
+        mDragItemAdapter = adapter;
+        mLayoutManager = layoutManager;
+        mItemDecorations = itemDecorations;
+        mHasFixedItemSize = hasFixedItemSize;
+        mColumnBackgroundColor = columnBackgroundColor;
+        mItemsSectionBackgroundColor = itemsSectionBackgroundColor;
+        mHeader = header;
+        mColumnDragView = columnDragView;
+    }
+
+    @NonNull DragItemAdapter getDragItemAdapter() {
+        return mDragItemAdapter;
+    }
+
+    LayoutManager getLayoutManager() {
+        return mLayoutManager;
+    }
+
+    @NonNull List<RecyclerView.ItemDecoration> getItemDecorations() {
+        return mItemDecorations;
+    }
+
+    boolean isHasFixedItemSize() {
+        return mHasFixedItemSize;
+    }
+
+    @ColorInt int getColumnBackgroundColor() {
+        return mColumnBackgroundColor;
+    }
+
+    @ColorInt int getItemsSectionBackgroundColor() {
+        return mItemsSectionBackgroundColor;
+    }
+
+    View getHeader() {
+        return mHeader;
+    }
+
+    View getColumnDragView() {
+        return mColumnDragView;
+    }
+
+    /**
+     * Builder for {@link ColumnProperties}.
+     */
+    public static class Builder {
+
+        private DragItemAdapter mDragItemAdapter;
+        private LayoutManager mLayoutManager = null;
+        private ArrayList<RecyclerView.ItemDecoration> mItemDecoration = new ArrayList<>();
+        private boolean mHasFixedItemSize = false;
+        private int mColumnBackgroundColor = Color.TRANSPARENT;
+        private int mItemsSectionBackgroundColor = Color.TRANSPARENT;
```

Comment:
Curious about this one, was this added to be able to have a different color of the header background? Since if no header is added mItemsSectionBackgroundColor will always overwrite the column color. Is it more logical to only have mColumnBackgroundColor and then you control the header background color from the actual header view that you choose to add yourself?

Reply:
I faced the case when I needed to have a transparent header background and colored background of the items list. I prefer to leave mItemsSectionBackgroundColor, but in my opinion, if we gather all the properties of a column to one file we should add the possibility to set the whole column color. Also, we can add a frame for the header and set colors to the header and to the items separately. What do you think about this?

Reply:
Ah I see, let's leave it like this.